website/pages/blog/announcing-cloudquery-python-sdk.mdx

```diff
@@ -0,0 +1,148 @@
+---
+title: Announcing the Python SDK for CloudQuery Plugin Development
+tag: announcement
+date: 2023/08/04
+description:
+author: hermanschaaf
+---
+
+import { BlogHeader } from "../../components/BlogHeader"
+
+<BlogHeader/>
+
+We're excited to announce the first release of a Python SDK for CloudQuery plugin development! This SDK provides a high-level toolkit for developing CloudQuery plugins in Python.
+
+## Background
+
+CloudQuery is designed with a pluggable architecture and uses [Apache Arrow](https://arrow.apache.org/) over [gRPC](https://grpc.io/) for communication between plugins. Source and destination plugins are independent of one another, and this architecture allows plugins to be written in different languages but still communicate with one another. Until now, we've only provided an SDK for writing plugins in Go, but we're excited to announce that we now have an SDK for writing plugins in Python as well.
+
+![cloudquery high-level architecture](/images/cloudquery-architecture.png)
+
+## Features
+
+The Python SDK provides a number of features to make plugin development easier, and plugins written in Python performant.
+
+### Plugin Server
+
+The most basic functionality provided by the Python SDK is to start a gRPC plugin server that supports all the flags expected by the CloudQuery CLI. This allows you to write a plugin in Python and run it using the same command line interface as any other plugin.
+
+The following example shows how to create a plugin server that runs a plugin called `MyPlugin`:
+
+```python
+import sys
+from cloudquery.sdk import serve
+
+from plugin import MyPlugin
+
+
+def main():
+    p = MyPlugin()
+    serve.PluginCommand(p).run(sys.argv[1:])
+
+
+if __name__ == "__main__":
+    main()
+```
+
+Here `MyPlugin` is a class that extends `cloudquery.sdk.plugin.Plugin`. We'll look at the `Plugin` class in more detail next.
+
+### Plugin Class
+
+A CloudQuery Python source plugin, like `MyPlugin` above, should subclass `cloudquery.sdk.plugin.Plugin` and needs to implement three methods: `init`, `get_tables` and `sync`. The `init` method is called when the plugin is started, and is where you can do any initialization work. The `get_tables` method should return a list of tables that the plugin supports. The `sync` method is called when a table needs to be synced, and is where the SDK scheduler can be used to manage the syncing of all the supported tables.
+
+### Multi-threaded Scheduler
+
+{/* <!-- vale off --> */}
+The scheduler's responsibility is to manage the order in which tables are synced, ensuring that tables are synced in the correct order to avoid dependency issues, and to place limits on the number of concurrent requests and memory usage. For the scheduler in the Python SDK, we implemented a new queue-based scheduling algorithm that can take advantage of multiple cores, without plugin developers needing to think about this.
```

Comment:
```suggestion
The scheduler's responsibility is to manage the order in which tables are synced in order to avoid dependency issues, and place limits on the number of concurrent requests and memory usage. For the scheduler in the Python SDK, we implemented a new queue-based scheduling algorithm that can take advantage of multiple cores, without plugin developers needing to think about this.
```